src/nvim/eval.c

```diff
@@ -15073,15 +15073,15 @@ static void set_buffer_lines(buf_T *buf, linenr_T lnum, typval_T *lines,
     }
 
     rettv->vval.v_number = 1;  // FAIL
-    if (line == NULL || lnum < 1 || lnum > curbuf->b_ml.ml_line_count + 1) {
```

Comment:
function exits on same condition, line 15034

Reply:
This is done in Vim already, via https://github.com/vim/vim/commit/1c17ffa4611f4efe68c61f7cdd9ed692a866ba75 (patch 8.0.1753: various warnings from a static analyser).
I think we should import that patch instead / additionally.

Reply:
This could also be done with this PR then, e.g. by running `scripts/vim-patch.sh -P 8.0.1753`, ensuring that all the fixes from there are applied, and using the commit message from the script/new branch then.

Reply:
ack

Reply:
@blueyed 
Using the script that you mentioned did not work for me, it was failing at every file so I went over the patch changes manually, and nothing else is applicable to nvim code. Most of the times I was not even able to find a relevant line.

Reply:
the script simply prepares a proper commit message, and produces a patch file. The patch file is for reference, it's not really intended to be automatically applied.

Reply:
@justinmk thanks.
I think that this is the only change that is applicable to nvim.
I have changed the commit message and uncommented the patch number. Is there anything else I need to do?

Reply:
src/nvim/version.c should not be changed.
Also the commit with the patch commit message only has linting?  I would have used it either as a separate commit with the fix you did, or just with the commit message for that commit already.

Reply:
The patch for getchar.c also applies - but others are not relevant (N/A), or have even been fixed via PVS before (2cdf9d138).